Core/Core/Discussions/GetDiscussions.swift

```diff
@@ -40,6 +40,26 @@ class GetAnnouncements: CollectionUseCase {
     ) }
 }
 
+public class GetAllAnnouncements: CollectionUseCase {
+    public typealias Model = DiscussionTopic
+
+    var contextCodes: [String]?
+
+    public var cacheKey: String? {
+        let codes = contextCodes?.joined(separator: ",") ?? ""
+        return "get-announcements-\(codes)"
+    }
+    public var request: GetAllAnnouncementsRequest
+    public var scope = Scope(predicate: .all, order: [
```

Comment:
You'll want to make sure you exclude any regular discussions that might be in the db.
```
.where(
    #keyPath(DiscussionTopic.isAnnouncement), equals: true,
    orderBy: #keyPath(DiscussionTopic.postedAt), ascending: false
)
```